### PR TITLE
Fix: e2e test failes due to webhook not ready

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

**Proposed changes:**
- Wait for the webhooks to be ready in e2e test CI

## Change Category
Indicate the type of change by marking the applicable boxes:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### Rationale

The [e2e CI](https://github.com/kubeflow/spark-operator/actions/runs/10678444394/job/29595500565) occasionally failed due to webhook not ready. 

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] Existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by jacobsalway** [comment]
Acking the review assignment for tomorrow and thanks for taking a look at this before me

File: test/e2e/suit_test.go [hunk 1]
```diff
@@ -53,6 +56,12 @@ import (
 const (
 	ReleaseName      = "spark-operator"
 	ReleaseNamespace = "spark-operator"
+
+	MutatingWebhookName   = "spark-operator-webhook"
+	ValidatingWebhookName = "spark-operator-webhook"
```
Comment (jacobsalway) [inline test/e2e/suit_test.go:61]: It looks like these constants aren't used where I imagine you intended to use them on lines 148 and 149.

File: test/e2e/suit_test.go
Comment (jacobsalway) [inline test/e2e/suit_test.go:1]: I like that you've pulled the `waitFor...` functions out into their own utilities. Should cut down on some boilerplate repetition if we add more e2e tests

File: test/e2e/suit_test.go (hunk 1 above)
Comment (ChenYi015) [reply test/e2e/suit_test.go:61]: You are right, I have fixed it.

**Comment by jacobsalway** [comment]
Hmm I've run it a few times locally and was able to replicate the `connection refused` error once but not after that. I wonder if it's flakey depending on the exact poll timing vs when the bundle was updated and adding some buffer time might help?

> After you create the webhook configuration, the system will take a few seconds to honor the new configuration.

https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#configure-admission-webhooks-on-the-fly

**Comment by ChenYi015** [comment]
> I added another check to ensure that webhook service endpoints are not empty.

I did 10+ e2e tests locally, and all of them complete successfully.

**Comment by jacobsalway** [comment]
Noting for posterity: we discussed this issue/change separately and think the flakiness is due to the `MutatingWebhookConfiguration` being created by Helm before the controller has generated the certs/the deployment or service objects are ready.

I'm happy to approve this PR as is/just add a time.Sleep buffer to fix the e2e flakiness if it's an issue, but we could also find a better fix

**Pull request merged**